go/service/gregor.go

```diff
@@ -220,26 +220,29 @@ func (g *gregorHandler) serverSync(ctx context.Context, cli gregor1.IncomingInte
 		if pt != nil {
 			t = *pt
 		}
+		g.G().Log.Debug("gregor handler: starting sync from: %s", t)
 	} else {
 		g.G().Log.Debug("gregor handler: performing a fresh sync")
 	}
 
 	// Sync down everything from the server
-	if err = g.gregorCli.Sync(cli); err != nil {
+	consumedMsgs, err := g.gregorCli.Sync(cli)
+	if err != nil {
 		g.G().Log.Errorf("gregor handler: error syncing from the server, bailing: %s", err)
-		return err
+		return []gregor.InBandMessage{}, []gregor.InBandMessage{}, err
```

Comment:
ditto everywhere else.  BTW in go you if you `len(nil)` you'll get `0`, and not a coredump.